Homework7/graphql integration

LiliaBilous opened the pull request:
## Summary

Changed the project structure by splitting the codebase into two main folders: `frontend` and `backend`.  
To reduce the number of changed files in this PR, all existing project files were first pushed to the `develop` branch in [PR#15](https://github.com/GenesisEducationKyiv/front-end-school-3-0-LiliaBilous/pull/15).

## Backend updates

- Implemented a GraphQL API instead of the existing REST API — Homework Part 1.
- Added a GraphQL Subscription that streams the currently active track in real-time. The active track changes randomly every 1–2 seconds on the server — Homework Part 2.

## Frontend updates

- Subscribed to the active track updates via GraphQL Subscription and rendered the currently active track title — Homework Part 2.
- Updated state management and data fetching to work with GraphQL.
- Applied layout and component style improvements.

## Notes

Commands for local testing:

- **Backend**
  ```
  cd backend
  npm install
  npm run dev
  ```

-  **Frontend**
    ```
    cd frontend
    npm install
    npm run dev
    ```

LiliaBilous committed:
feat: add GraphQL setup and backend logic for active track, install dependencies

LiliaBilous committed:
feat: integrate Apollo Client with GraphQL subscriptions and add ActiveTrack component

LiliaBilous committed:
chore: update .gitignore to exclude data directories and remove .DS_Store

LiliaBilous committed:
feat: update CSS variables for improved theming and add transition animations

LiliaBilous committed:
feat: enhance modal and genre selector styles for improved UI consistency

LiliaBilous committed:
feat: refactor AppFooter and PaginationControls for improved structure and styling

LiliaBilous committed:
feat: enhance FilterTabGroup and TrackToolbar for improved UI and accessibility

LiliaBilous committed:
feat: enhance UI components and styles for improved consistency and accessibility

LiliaBilous committed:
feat: enhance ActiveTrack component with transition effects and improved styling

LiliaBilous committed:
feat: add favicon

LiliaBilous committed:
feat: added graphQL api

LiliaBilous committed:
feat: add genres and trackBySlug queries to GraphQL schema and resolvers

LiliaBilous committed:
feat: migrate audio and track services to GraphQL, update schemas and store logic

LiliaBilous committed:
feat: add slug generation for tracks and enforce uniqueness on creation

LiliaBilous committed:
chore: update dependencies

LiliaBilous committed:
fix: correct import path for getTrackBySlug and adjust GraphQL query parameters

alyanoyigor commented on frontend/src/features/audio/store/audioStore.ts:
```diff
@@ -17,6 +17,7 @@ export const useTrackAudioStore = defineStore('trackAudioStore', () => {
       }
       trackBySlug.value = fullTrack
     }
+    console.log(result)
```
Це напевно краще видалити

LiliaBilous replied:
так згодна, видалила у домашній №8...у цій залишилось

LiliaBilous commented:
### Additional Fixes

I made a few technical improvements after merging the initial PR:

- Fixed a bug in trackFilterStore.ts where the page query param was treated as a string, preventing correct pagination.
- Updated REST API file URLs to use API_URL instead of hardcoded values to avoid issues without env variables.
- Replaced in-place array mutation in useTrackFileStore.ts with an immutable update to ensure correct reactivity.

These changes address minor issues and reduce technical debt.
